internal/entrypoint/run.go

```diff
@@ -181,7 +186,13 @@ func Run(ctx context.Context, config *Config, exporter otlexporters.Otlexporter,
 			logger.WithField("number of PowerFlexClient", len(config.PowerFlexClient)).Debug("PowerFlexClient")
 
 			config.Lock.Lock()
-			for key, sioConfig := range config.PowerFlexClient {
+			powerflexClients := make(map[string]*PowerflexClient)
+			for k, v := range config.PowerFlexClient {
+				powerflexClients[k] = v
+			}
+			config.Lock.Unlock()
+
+			for key, sioConfig := range powerflexClients {
 				logger.WithField("storage system id", key).Debug("storage system id")
```

Comment:
You are repeating the same code in multiple places. This is a good candidate for a small utility function to return the copy for you.